### ISSUE TITLE
no-async-in-computed-properties fails on objects containing async functions

### ISSUE THREAD

**Issue by cjpearson** [vuejs/eslint-plugin-vue#725]
This can be seen on the demo page for the rule [here](https://vuejs.github.io/eslint-plugin-vue/rules/no-async-in-computed-properties.html). Adding the following computed property causes an ESLint error.

```vue
test () {
  return {
    callback: async (e) => e
  }
}
```

**What did you expect to happen?**
No error since computing the result of the computed property is not asynchronous.

**What actually happened?**
There is an error on the property.
